images/decode.go

```diff
@@ -153,6 +157,12 @@ func IsWebp(buf []byte) bool {
 		buf[10] == 0x42 && buf[11] == 0x50
 }
 
+func IsIco(buf []byte) bool {
+	return len(buf) > 4 &&
+		buf[0] == 0 && buf[2] == 1 || buf[2] == 2 &&
+		buf[4] > 0
+}
```

Comment:
According to Identification section [here](http://fileformats.archiveteam.org/wiki/ICO)